File: Extras/primo.cpp
#include <iostream>

using namespace std;

int main(int argc, char* argv[]) {
	int n;

	do {
		cout << "Dame un número: ";
		cin >> n;
		if(n < 1) {
			cout << "Número inválido";
		}
	} while(n < 1);

	if (n == 1) {
		cout << n << " NO es un número primo." << endl;
	} else {
		for (int i=2; i <= n; i++) {
			if (n % i == 0) {
				if (n != i) {
					cout << n << " NO es un número primo." << endl;
					return 0;
				} else {
					cout << n << " Si es un número primo." << endl;
				}
			}
		}
	}


	return 0;
}

File: Extras/primo.py
# -*- coding: utf-8 -*-

n = 0
while n < 1:
    n = int(raw_input("Dame un número: "))
    if n < 1:
        print "Número inválido"

if n == 1:
    print str(n) + " NO es un número primo"
else:
    for i in range(2, n+1):
        if n % i == 0:
            if n != i:
                print str(n) + " NO es un número primo"
                break
            else:
                print str(n) + " SI es un número primo"